src/main/java/org/example/server/MySocketServer.java

```diff
@@ -6,57 +6,70 @@
 import java.io.OutputStream;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class MySocketServer {
+    private static ExecutorService pool;
     private static final String END_OF_MESSAGE_MARK = "\n";
-    private static final String EXIT_MESSAGE = "Q";
 
-    public static void main(String[] args) throws IOException {
+    public static void main(String[] args) {
+        final int MAX_CONNECTIONS = 10;
         final int PORT = 9999;
 
-        startServer(PORT);
+        startServer(MAX_CONNECTIONS, PORT);
     }
 
-    private static void startServer(int port) throws IOException {
-        BufferedReader in = null;
-        OutputStream out = null;
-        String receivedMessage;
+    private static void startServer(int maxConnections, int port) {
+        pool = Executors.newFixedThreadPool(maxConnections);
 
         try (ServerSocket serverSocket = new ServerSocket(port)) {
             System.out.println("Server started on port: " + port);
+
+            while (true) {
+                listenForClients(serverSocket);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void listenForClients(ServerSocket serverSocket) {
+        try {
             Socket clientSocket = serverSocket.accept();
             System.out.println("Client connected");
 
-            in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
-            out = clientSocket.getOutputStream();
+            pool.submit(() -> handleClient(clientSocket));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
 
-            while (true) {
+    private static void handleClient(Socket clientSocket) {
+        try (
+                BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+                OutputStream out = clientSocket.getOutputStream()
+        ) {
+            String receivedMessage = in.readLine();
+            if (receivedMessage != null) {
+                System.out.println("Client NikName: " + receivedMessage);
+            }
```

Comment:
@korvalanchik something similar is here as well. There is no need to read a message from the client twice (first one before the `while` loop, second one - inside the loop). So please consider to use below snippet of the code
`        try (
                BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
                OutputStream out = clientSocket.getOutputStream()
        ) {
            while(true) {
                String receivedMessage = in.readLine();
                if (receivedMessage != null) {
                    System.out.println("Client NikName: " + receivedMessage);
                }
                if (receivedMessage != null) {
                    String serverResponse = receivedMessage.toUpperCase() + END_OF_MESSAGE_MARK;
                    out.write(serverResponse.getBytes());
                    out.flush();
                }
            }
        } catch (IOException e) {
`